lib/core/api.js

```diff
@@ -359,8 +359,13 @@ function addDataToDataset(data, datasetId, blockIndex, uploadType, callback) {
       callback('duplicate', jqxhr.responseJSON.index);
     } else {
       api.log('platform add data to dataset failed.');
-      api.log(status);
-      callback(status);
+      var err = new Error('Uploading data to platform failed.');
+      err.name = 'API Error';
+      err.status = status;
+      err.datasetId = datasetId;
+      err.sessionId = status.meta.trace.request;
```

Comment:
@darinkrauss Will you make a card on the backlog to add the session ID so that we don't forget?